SignalServiceKit/src/Storage/TSStorageManager.m

```diff
@@ -514,20 +515,26 @@ - (NSData *)databasePassword
         [SAMKeychain passwordForService:keychainService account:keychainDBPassAccount error:&keyFetchError];
 
     if (keyFetchError) {
-        // FIXME SHARINGEXTENSION
-        //        UIApplicationState applicationState = [UIApplication sharedApplication].applicationState;
-        //        NSString *errorDescription = [NSString stringWithFormat:@"Database password inaccessible. No unlock
-        //        since device restart? Error: %@ ApplicationState: %d", keyFetchError, (int)applicationState];
-        //        DDLogError(@"%@ %@", self.logTag, errorDescription);
-        //        [DDLog flushLog];
-
-        // FIXME SHARINGEXTENSION
-        //        if ([UIApplication sharedApplication].applicationState == UIApplicationStateBackground) {
-        //            // TODO: Rather than crash here, we should detect the situation earlier
-        //            // and exit gracefully - (in the app delegate?). See the `
-        //            // This is a last ditch effort to avoid blowing away the user's database.
-        //            [self backgroundedAppDatabasePasswordInaccessibleWithErrorDescription:errorDescription];
-        //        }
+        NSString *errorDescription =
+            [NSString stringWithFormat:@"Database password inaccessible. No unlock since device restart? Error: %@",
+                      keyFetchError];
+        if (CurrentAppContext().isMainApp) {
+            UIApplicationState applicationState = CurrentAppContext().mainApplicationState;
+            errorDescription =
+                [errorDescription stringByAppendingFormat:@", ApplicationState: %d", (int)applicationState];
+        }
+        DDLogError(@"%@ %@", self.logTag, errorDescription);
+        [DDLog flushLog];
+
+        if (CurrentAppContext().isMainApp) {
+            UIApplicationState applicationState = CurrentAppContext().mainApplicationState;
+            if (applicationState == UIApplicationStateBackground) {
+                // TODO: Rather than crash here, we should detect the situation earlier
+                // and exit gracefully - (in the app delegate?). See the `
+                // This is a last ditch effort to avoid blowing away the user's database.
+                [self backgroundedAppDatabasePasswordInaccessibleWithErrorDescription:errorDescription];
+            }
+        }
```

Comment:
Do we want to:

```
else { 
   throw exception
}
```

I don't think the share extension should ever be setting the db password, should it?

Reply:
Done, good point.